qless-java/src/main/java/com/moz/qless/JobSpec.java

```diff
@@ -0,0 +1,94 @@
+package com.moz.qless;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.moz.qless.client.ClientHelper;
+
+public class JobSpec {
+
+  protected String klass;
```

Comment:
Yes, my inclination is that if you do plan on having inheritance, you shouldn't rely on a member variable.